### PR TITLE
[#8862] Use extras in tox.ini

### PR DESCRIPTION
**Pull request by derwolfe**
https://twistedmatrix.com/trac/ticket/8862#ticket

### CODE REVIEW

**Comment by derwolfe** [comment]
It looks like these are failing because of `gmpy`.

**Comment by markrwilliams** [comment]
@derwolfe This can either wait til #540 gets merged, or you change the `all_non_platform` extra to not imply `gmpy`.

**Comment by derwolfe** [comment]
@markrwilliams I'd be ok waiting :) Thanks for the heads up!

File: tox.ini [hunk 1]
```diff
@@ -39,31 +39,13 @@ envlist=
...
-    {alldeps,tests,coverage}: h2
-    {alldeps,tests,coverage}: priority
-    {alldeps,tests,coverage}: sphinx
+    {alldeps,tests,coverage}: twisted[all_non_platform]
 
     py27-{alldeps,tests,coverage}-{posix,macos}: pysqlite
-    py27-{alldeps,tests,coverage}: soappy
 
-    windows: pypiwin32
+    windows: twisted[windows_platform]
```
Comment (derwolfe) [inline tox.ini]: The more I think about this, the more I think that this should be `.[windows_platform]`. Verifying this should be pretty simple.

**Comment by derwolfe** [comment]
Thanks @rodrigc 😄

**Review by rodrigc — CHANGES_REQUESTED** [review]
This change doesn't work on OS X.

This doesn't work:

`/Users/crodrigues/twisted8/build/py36-alldeps-withcov-macos/bin/pip install '.[osx_platform]'`
Processing /Users/crodrigues/twisted8
Exception:
Traceback (most recent call last):
  File "/Users/crodrigues/twisted8/build/py36-alldeps-withcov-macos/lib/python3.6/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Users/crodrigues/twisted8/build/py36-alldeps-withcov-macos/lib/python3.6/site-packages/pip/commands/install.py", line 310, in run
    wb.build(autobuilding=True)
  File "/Users/crodrigues/twisted8/build/py36-alldeps-withcov-macos/lib/python3.6/site-packages/pip/wheel.py", line 750, in build
    self.requirement_set.prepare_files(self.finder)
  File "/Users/crodrigues/twisted8/build/py36-alldeps-withcov-macos/lib/python3.6/site-packages/pip/req/req_set.py", line 370, in prepare_files
    ignore_dependencies=self.ignore_dependencies))
  File "/Users/crodrigues/twisted8/build/py36-alldeps-withcov-macos/lib/python3.6/site-packages/pip/req/req_set.py", line 587, in _prepare_file
    session=self.session, hashes=hashes)
  File "/Users/crodrigues/twisted8/build/py36-alldeps-withcov-macos/lib/python3.6/site-packages/pip/download.py", line 798, in unpack_url
    unpack_file_url(link, location, download_dir, hashes=hashes)
  File "/Users/crodrigues/twisted8/build/py36-alldeps-withcov-macos/lib/python3.6/site-packages/pip/download.py", line 676, in unpack_file_url
    shutil.copytree(link_path, location, symlinks=True)
  File "/Users/crodrigues/twisted8/build/py36-alldeps-withcov-macos/lib/python3.6/shutil.py", line 353, in copytree
    raise Error(errors)
shutil.Error: [('/Users/crodrigues/twisted8/_trial_temp/twisted.internet.test.test_unix/UNIXDatagramTestsBuilder_Asyncio/test_listenMode/_ehk9xjx/temp', '/var/folders/m0/xq5mbjy11b175l99w_1d_5ww0000gn/T/pip-1po87yis-build/_trial_temp/twisted.internet.test.test_unix/UNIXDatagramTestsBuilder_Asyncio/test_listenMode/_ehk9xjx/temp', "[Errno 102] Operation not supported on socket: '/Users/crodrigues/twisted8/_trial_temp/twisted.internet.test.test_unix/UNIXDatagramTestsBuilder_Asyncio/test_listenMode/_ehk9xjx/temp'"), ('/Users/crodrigues/twisted8/_trial_temp/twisted.internet.test.test_unix/UNIXDatagramTestsBuilder_SelectR/test_listenMode/9b5xw2h0/temp', '/var/folders/m0/xq5mbjy11b175l99w_1d_5ww0000gn/T/pip-1po87yis-build/_trial_temp/twisted.internet.test.test_unix/UNIXDatagramTestsBuilder_SelectR/test_listenMode/9b5xw2h0/temp', "[Errno 102] Operation not supported on socket: '/Users/crodrigues/twisted8/_trial_temp/twisted.internet.test.test_unix/UNIXDatagramTestsBuilder_SelectR/test_listenMode/9b5xw2h0/temp'"), ('/Users/crodrigues/twisted8/_trial_temp/twisted.internet.test.test_unix/UNIXTestsBuilder_AsyncioSelector/test_mode/jrnppzyv/temp', '/var/folders/m0/xq5mbjy11b175l99w_1d_5ww0000gn/T/pip-1po87yis-build/_trial_temp/twisted.internet.test.test_unix/UNIXTestsBuilder_AsyncioSelector/test_mode/jrnppzyv/temp', "[Errno 102] Operation not supported on socket: '/Users/crodrigues/twisted8/_trial_temp/twisted.internet.test.test_unix/UNIXTestsBuilder_AsyncioSelector/test_mode/jrnppzyv/temp'"), ('/Users/crodrigues/twisted8/_trial_temp/twisted.internet.test.test_unix/UNIXTestsBuilder_SelectReactorTe/test_mode/zkdr575o/temp', '/var/folders/m0/xq5mbjy11b175l99w_1d_5ww0000gn/T/pip-1po87yis-build/_trial_temp/twisted.internet.test.test_unix/UNIXTestsBuilder_SelectReactorTe/test_mode/zkdr575o/temp', "[Errno 102] Operation not supported on socket: '/Users/crodrigues/twisted8/_trial_temp/twisted.internet.test.test_unix/UNIXTestsBuilder_SelectReactorTe/test_mode/zkdr575o/temp'")]

but this works:

`/Users/crodrigues/twisted8/build/py36-alldeps-withcov-macos/bin/pip install 'twisted[osx_platform]'`
Requirement already satisfied (use --upgrade to upgrade): twisted[osx_platform] in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages
Requirement already satisfied (use --upgrade to upgrade): zope.interface>=4.0.2 in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages (from twisted[osx_platform])
Requirement already satisfied (use --upgrade to upgrade): idna>=0.6; extra == "osx_platform" in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages (from twisted[osx_platform])
Requirement already satisfied (use --upgrade to upgrade): appdirs>=1.4.0; extra == "osx_platform" in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages (from twisted[osx_platform])
Requirement already satisfied (use --upgrade to upgrade): pyserial; extra == "osx_platform" in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages (from twisted[osx_platform])
Requirement already satisfied (use --upgrade to upgrade): pyopenssl>=16.0.0; extra == "osx_platform" in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages (from twisted[osx_platform])
Requirement already satisfied (use --upgrade to upgrade): pyasn1; extra == "osx_platform" in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages (from twisted[osx_platform])
Collecting pyobjc; extra == "osx_platform" (from twisted[osx_platform])
  Using cached pyobjc-3.1.1.tar.gz
Requirement already satisfied (use --upgrade to upgrade): h2<3.0,>=2.3.0; extra == "osx_platform" in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages (from twisted[osx_platform])
Requirement already satisfied (use --upgrade to upgrade): cryptography>=0.9.1; extra == "osx_platform" in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages (from twisted[osx_platform])
Requirement already satisfied (use --upgrade to upgrade): gmpy; extra == "osx_platform" in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages (from twisted[osx_platform])
Requirement already satisfied (use --upgrade to upgrade): soappy; extra == "osx_platform" in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages (from twisted[osx_platform])
Requirement already satisfied (use --upgrade to upgrade): priority<2.0,>=1.1.0; extra == "osx_platform" in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages (from twisted[osx_platform])
Requirement already satisfied (use --upgrade to upgrade): service-identity; extra == "osx_platform" in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages (from twisted[osx_platform])
Requirement already satisfied (use --upgrade to upgrade): setuptools in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages (from zope.interface>=4.0.2->twisted[osx_platform])
Requirement already satisfied (use --upgrade to upgrade): six>=1.5.2 in ./build/py36-alldeps-withcov-macos/lib/python3.6/site-packages (from pyopenssl>=16.0.0; extra == "osx_platform"->twisted[osx_platform])
Collecting py2app>=0.10 (from pyobjc; extra == "osx_platform"->twisted[osx_platform])
  Using cached py2app-0.10.tar.gz
Collecting pyobjc-core==3.1.1 (from pyobjc; extra == "osx_platform"->twisted[osx_platform])
Collecting pyobjc_framework-AVKit==3.1.1 (from pyobjc; extra == "osx_platform"->twisted[osx_platform])
  Using cached pyobjc-framework-AVKit-3.1.1.tar.gz

**Pull request closed without merging**